### PR TITLE
[7.x] Fix ChromeDriverCommand for ChromeDriver 115+

### LINKED ISSUE #1040: Downloading ChromeDriver 115+ fails
### Dusk Version

7.8.0

### Laravel Version

9.52.10

### PHP Version

8.1.16

### PHPUnit Version

9.6.10

### Database Driver & Version

_No response_

### Description

Dusk fails to download ChromeDriver 115+ because Google changed the URL's.

> Starting with M115 the latest Chrome + ChromeDriver releases per release channel (Stable, Beta, Dev, Canary) are available at[ the Chrome for Testing availability dashboard](https://googlechromelabs.github.io/chrome-for-testing/). For automated version downloading one can use the convenient [JSON endpoints](https://googlechromelabs.github.io/chrome-for-testing/last-known-good-versions-with-downloads.json).
>
> https://chromedriver.chromium.org/home

### Steps To Reproduce

Run `php artisan dusk:chrome-driver 115.0.5790.98`

```
  ErrorException 

  file_get_contents(https://chromedriver.storage.googleapis.com/115.0.5790.98/chromedriver_linux64.zip): Failed to open stream: HTTP request failed! HTTP/1.1 404 Not Found

  at vendor/laravel/dusk/src/Console/ChromeDriverCommand.php:322
     318▕         }
     319▕
     320▕         $streamContext = stream_context_create($contextOptions);
     321▕
  -> 322▕         return file_get_contents($url, false, $streamContext);
     323▕     }
     324▕ }
     325▕

      +16 vendor frames 
  17  artisan:37
      Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
```

### PR DESCRIPTION
**Pull request by crynobone**
fixes #1040

### CODE REVIEW

File: src/Console/ChromeDriverCommand.php [hunk 1]
```diff
@@ -30,40 +32,6 @@ class ChromeDriverCommand extends Command
...
-     * @var string
-     */
-    protected $versionUrl = 'https://chromedriver.storage.googleapis.com/LATEST_RELEASE_%d';
-
-    /**
-     * URL to the ChromeDriver download.
-     *
-     * @var string
-     */
-    protected $downloadUrl = 'https://chromedriver.storage.googleapis.com/%s/chromedriver_%s.zip';
```
Comment (crynobone) [inline src/Console/ChromeDriverCommand.php:52]: URLs will be hardcoded based on usage to support different version usage.

File: src/Console/ChromeDriverCommand.php [hunk 2]
```diff
@@ -30,40 +32,6 @@ class ChromeDriverCommand extends Command
...
-     *
-     * @var array
-     */
-    protected $slugs = [
-        'linux' => 'linux64',
-        'mac' => 'mac64',
-        'mac-intel' => 'mac64',
-        'mac-arm' => 'mac_arm64',
-        'win' => 'win32',
-    ];
```
Comment (crynobone) [inline src/Console/ChromeDriverCommand.php:65]: Moved to `OperatingSystem`, use `OperatingSystem::chromeDriverSlug($os)`.

File: src/Console/ChromeDriverCommand.php [hunk 3]
```diff
@@ -106,29 +74,6 @@ class ChromeDriverCommand extends Command
...
-        'mac-intel' => [
-            '/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome --version',
-        ],
-        'mac-arm' => [
-            '/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome --version',
-        ],
-        'win' => [
-            'reg query "HKEY_CURRENT_USER\Software\Google\Chrome\BLBeacon" /v version',
-        ],
-    ];
```
Comment (crynobone) [inline src/Console/ChromeDriverCommand.php:130]: Moved to `OperatingSystem`, use `OperatingSystem::chromeVersionCommands($os)`.

File: src/OperatingSystem.php [hunk 4]
```diff
@@ -3,9 +3,107 @@
...
+            'commands' => [
+                '/Applications/Google\ Chrome\ for\ Testing.app/Contents/MacOS/Google\ Chrome\ for\ Testing --version',
+                '/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome --version',
+            ],
+        ],
+        'mac-arm' => [
+            'slug' => 'mac-arm64',
+            'commands' => [
+                '/Applications/Google\ Chrome\ for\ Testing.app/Contents/MacOS/Google\ Chrome\ for\ Testing --version',
+                '/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome --version',
```
Comment (crynobone) [inline src/OperatingSystem.php:43]: Add support for `Chrome for Testing.app`, available from version `113`: https://googlechromelabs.github.io/chrome-for-testing/latest-versions-per-milestone-with-downloads.json

File: src/Console/ChromeDriverCommand.php [hunk 5]
```diff
@@ -245,36 +183,41 @@ protected function detectChromeVersion($os)
...
-            $this->getUrl($url)
-        );
+        $resource = Utils::tryFopen($archive = $this->directory.'chromedriver.zip', 'w');
+
+        Http::withOptions(array_merge([
+            'verify' => $this->option('ssl-no-verify') === false,
+            'sink' => $resource,
+        ]), array_filter([
+            'proxy' => $this->option('proxy'),
+        ]))->get($url);
```
Comment (crynobone) [inline src/Console/ChromeDriverCommand.php:200]: Use Laravel HTTP Client to speed up downloading ChromeDriver 115.

File: composer.json [hunk 6]
```diff
@@ -13,10 +13,11 @@
         "php": "^8.0",
         "ext-json": "*",
         "ext-zip": "*",
-        "php-webdriver/webdriver": "^1.9.0",
-        "nesbot/carbon": "^2.0",
+        "guzzlehttp/guzzle": "^7.2",
```
Comment (crynobone) [inline composer.json:16]: Allows to use HTTP Client.

**Review by driesvints — APPROVED** [review]
Looks mior!

**Pull request merged**